### PR TITLE
feature/Converting Js file to Typescript

### PR DESCRIPTION
**Pull request by Liam-CurmideGray**

### CODE REVIEW

File: src/Inquirer.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import clear = require("clear");
+import { StartAnimation, StopAnimation } from "./loaderAnimation";
+import {
+  StarwarsModule,
+  StarshipValue,
+  PilotValue,
+  StarshipAnswer,
+  PilotAnswer
+} from "./starwars.model";
+import { Spinner } from "clui";
```
Comment (matthew-bettingcloud) [inline src/Inquirer.ts]: move up to line 4

File: src/Inquirer.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import inquirer from "inquirer";
+import { get as Get } from "./http.service";
+import clear = require("clear");
+import { StartAnimation, StopAnimation } from "./loaderAnimation";
```
Comment (matthew-bettingcloud) [inline src/Inquirer.ts]: add a new line to differentiate between external and internal modules

File: src/Inquirer.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    });
+}
+
+export function StarwarsQuestions(
+  starshipsResponse: StarwarsModule<StarshipValue>[],
+  buffer: Spinner
+): inquirer.ListQuestion<StarshipAnswer & PilotAnswer>[] {
+  StopAnimation(buffer);
+
+  let starshipQuestion: inquirer.Question<PilotAnswer> = {
```
Comment (matthew-bettingcloud) [inline src/Inquirer.ts]: these can be const instead of let?

File: src/http.service.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import axios from "axios";
+
+export async function get(url: string) {
```
Comment (matthew-bettingcloud) [inline src/http.service.ts]: missing return type

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { get as Get } from "./http.service";
+import { InquirerQ } from "./inquirer";
+import { StarwarsModule, StarshipValue } from "./starwars.model";
+import { TitleScreen } from "./titleScreen";
+import { StartAnimation } from "./loaderAnimation";
+import { Spinner } from "clui";
+
+TitleScreen();
+let buffer: Spinner = StartAnimation();
```
Comment (matthew-bettingcloud) [inline src/index.ts]: can be const?

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { get as Get } from "./http.service";
+import { InquirerQ } from "./inquirer";
+import { StarwarsModule, StarshipValue } from "./starwars.model";
+import { TitleScreen } from "./titleScreen";
+import { StartAnimation } from "./loaderAnimation";
+import { Spinner } from "clui";
```
Comment (matthew-bettingcloud) [inline src/index.ts]: formatting: differntiate between external imports and internal

example
```
import { Spinnger }  from "clui";

import { get } from "./http.service";
...etc
```

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import { get as Get } from "./http.service";
```
Comment (matthew-bettingcloud) [inline src/index.ts]: do you need it renamed to Get?

File: src/loaderAnimation.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { Spinner } from "clui";
+
+export function StartAnimation() {
```
Comment (matthew-bettingcloud) [inline src/loaderAnimation.ts]: functions are camelCase

File: src/titleScreen.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import figlet from "figlet";
+import chalk from "chalk";
+
```
Comment (matthew-bettingcloud) [inline src/titleScreen.ts]: remove extra line

File: src/titleScreen.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+export function TitleScreen(): void {
```
Comment (matthew-bettingcloud) [inline src/titleScreen.ts]: camelCase

File: src/titleScreen.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    
+    figlet(`Starwars`, function(err: Error | null, result?: string | undefined): void {
```
Comment (matthew-bettingcloud) [inline src/titleScreen.ts]: you can change it to

figlet(`Starwars`, (err, result) => {....

File: src/titleScreen.ts [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+    if(err != null) {
+        console.log("Something went wrong");
+        console.dir(err.message);
+
```
Comment (matthew-bettingcloud) [inline src/titleScreen.ts]: remove empty line

File: src/titleScreen.ts [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+    
+    figlet(`Starwars`, function(err: Error | null, result?: string | undefined): void {
+    if(err != null) {
+        console.log("Something went wrong");
+        console.dir(err.message);
+       
+    }
+
+    console.log(chalk.red.bold(result || "No data"));
+});
```
Comment (matthew-bettingcloud) [inline src/titleScreen.ts]: formatting?

File: bin/index.js [hunk 14]
```diff
@@ -1,88 +1,3 @@
...
-      name: pilot.data.name,
-      value: {
-        name: pilot.data.name,
-        height: pilot.data.height
-      }
-    });
-  }
-  return array;
-}
+require("../dist/index.js");
```
Comment (matthew-bettingcloud) [inline bin/index.js]: not sure about it (just curious) can you see if the below works? 
```
require("../dist");
```

File: package.json [hunk 15]
```diff
@@ -22,6 +23,16 @@
   "homepage": "https://github.com/Liam-CurmideGray/Git-RosettaStone#readme",
   "dependencies": {
     "axios": "^0.19.0",
-    "inquirer": "^6.4.1"
+    "chalk": "^2.4.2",
+    "clear": "^0.1.0",
+    "clui": "^0.3.6",
+    "figlet": "^1.2.3",
+    "inquirer": "^6.4.1",
+    "typescript": "^3.5.3"
```
Comment (matthew-bettingcloud) [inline package.json]: typescript is a dev dependency not dependency

File: src/Inquirer.ts [hunk 16]
```diff
@@ -0,0 +1,100 @@
+import inquirer from "inquirer";
+import { get as Get } from "./http.service";
```
Comment (matthew-bettingcloud) [inline src/Inquirer.ts]: leave it as `get` not `Get`

File: src/index.ts [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+
+async function getStarships(): Promise<StarwarsModule<StarshipValue>[]> {
+  let starshipUrl = `https://swapi.co/api/starships/`;
+  let starshipDetails: StarwarsModule<StarshipValue>[] = [];
+
+  while (starshipUrl != null) {
+    const starshipsApi = await Get(starshipUrl);
+
+    for (let i = 0; i < starshipsApi.data.results.length; i++) {
+      let starshipName: string = starshipsApi.data.results[i].name;
```
Comment (matthew-bettingcloud) [inline src/index.ts]: change it to this:
```
      let name: string = starshipsApi.data.results[i].name;
      let pilotUrls: string[] = starshipsApi.data.results[i].pilots;

      starshipDetails.push({
        name,
        value: {
          name,
          pilotUrls
        }
      });
```

File: src/starwars.model.ts [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+  name: string;
+  pilotUrls: string[];
+}
+
+export interface PilotValue {
+  name: string;
+  height: number;
+}
+
+export interface ShipAnswers {
```
Comment (matthew-bettingcloud) [inline src/starwars.model.ts]: `answer` or `Answers`?

File: bin/index.js (hunk 14 above)
Comment (Liam-CurmideGray) [reply bin/index.js]: Yes it works

**Pull request merged**